gpflow/utilities/utilities.py

```diff
@@ -40,6 +42,20 @@ def to_default_float(x):
     return tf.cast(x, dtype=default_float())
 
 
+class ShapeError(AssertionError):
+    pass
+
+
+def assert_shapes(shapes, *args, **kwargs):
```

Comment:
can we keep the argument names and types? It will at least help with autocompletion in the editor

Reply:
what if tensorflow just goes and changes them ? (not for our code, but if someone uses this part of gpflow with extra arguments that we're not testing for, and suddenly it won't work anymore?)

Reply:
i guess they could add further kwargs and be backwards compatible. For any other changes, i'd be worried we're not pinning our version of tensorflow tightly enough. if tensorflow's changing their API under our feet that's a problem, as our own usage of tf.function might be broken

i won't block the PR on this